codalab/lib/bundle_cli.py

```diff
@@ -1939,8 +1939,21 @@ def do_macro_command(self, args):
         if not getattr(args, metadata_util.metadata_key_to_argument('name')):
             setattr(args, metadata_util.metadata_key_to_argument('name'), 'new')
         # Reduce to the mimic case
-        args.bundles = [args.macro_name + '-in' + str(i+1) for i in range(len(args.bundles))] + \
-                       [args.macro_name + '-out'] + args.bundles
+        named_inputs = [bundle for bundle in args.bundles if ':' in bundle]
+        named_input_names = [bundle.split(':',1)[0] for bundle in named_inputs]
+
+        named_user_inputs = [bundle.split(':',1)[1] for bundle in named_inputs]
```

Comment:
Might be better to combine this with previous line and use `named_input_names, named_user_inputs = zip(*[...])`

Reply:
The zip approach doesn't work if there are no named inputs unless wrapped in another if block. Instead I just looped over all the input bundles and did all the processing bundle-by-bundle